src/Private/Console.ps1

```diff
@@ -0,0 +1,1259 @@
+            Write-PodeHostDivider -Force $true
+
+            $openAPIHeader = $true
+        }
+
+        # Print definition title
+        Write-PodeHost " '$key':" -ForegroundColor $titleColor -Force:$Force
+
+        # Determine endpoints for specification and documentation
+        if ($bookmarks.route.count -gt 1 -or $bookmarks.route.Endpoint.Name) {
+            # Directly use $bookmarks.route.Endpoint
+            Write-PodeHost "   $($PodeLocale.specificationMessage):" -ForegroundColor $subtitleColor -Force:$Force
+            foreach ($endpoint in $bookmarks.route.Endpoint) {
+                Write-PodeHost "     . $($endpoint.Protocol)://$($endpoint.Address)$($bookmarks.openApiUrl)" -ForegroundColor $urlColor -Force:$Force
+            }
+            Write-PodeHost "   $($PodeLocale.documentationMessage):" -ForegroundColor $subtitleColor -Force:$Force
+            foreach ($endpoint in $bookmarks.route.Endpoint) {
+                Write-PodeHost "     . $($endpoint.Protocol)://$($endpoint.Address)$($bookmarks.path)" -ForegroundColor $urlColor -Force:$Force
+            }
+        }
+        else {
+            # Use EndpointsInfo for fallback
+            Write-PodeHost "   $($PodeLocale.specificationMessage):" -ForegroundColor $subtitleColor -Force:$Force
+            $PodeContext.Server.EndpointsInfo | ForEach-Object {
+                if ($_.Pool -eq 'web') {
+                    $url = [System.Uri]::new([System.Uri]::new($_.Url), $bookmarks.openApiUrl)
+                    Write-PodeHost "     - $url" -ForegroundColor $urlColor -Force:$Force
+                }
+            }
+            Write-PodeHost "   $($PodeLocale.documentationMessage):" -ForegroundColor $subtitleColor -Force:$Force
+            $PodeContext.Server.EndpointsInfo | ForEach-Object {
+                if ($_.Pool -eq 'web') {
+                    $url = [System.Uri]::new([System.Uri]::new($_.Url), $bookmarks.path)
+                    Write-PodeHost "     - $url" -ForegroundColor $urlColor -Force:$Force
+                }
+            }
+        }
+    }
+    if ($openAPIHeader) {
+        # Footer
+
+        # Write a horizontal divider line to the console.
+        Write-PodeHostDivider -Force $true
+    }
+}
+
+<#
+.SYNOPSIS
+    Clears any remaining keys in the console input buffer.
+
+.DESCRIPTION
+    The `Clear-PodeKeyPressed` function checks if there are any keys remaining in the input buffer
+    and discards them, ensuring that no leftover key presses interfere with subsequent reads.
+
+.EXAMPLE
+    Clear-PodeKeyPressed
+    [Console]::ReadKey($true)
+
+    This example clears the buffer and then reads a new key without interference.
+
+.NOTES
+    This function is useful when using `[Console]::ReadKey($true)` to prevent previous key presses
+    from affecting the input.
+
+#>
+function Clear-PodeKeyPressed {
+    # Clear any remaining keys in the input buffer
+    while ([Console]::KeyAvailable) {
```

Comment:
For most cases this should be covered, but like `Get-PodeConsoleKey` you want to check `[Console]::IsInputRedirected`:

```powershell
while (![Console]::IsInputRedirected -and [Console]::KeyAvailable) {
    ...
}
```